src/query/management/src/serde/pb_serde.rs

```diff
@@ -56,3 +63,45 @@ where
 
     Ok(v)
 }
+
+pub async fn check_and_upgrade_to_pb<'a, T, ErrFn, CtxFn, D>(
+    key: String,
+    seq_value: &'a SeqV,
+    kv_api: Arc<dyn kvapi::KVApi<Error = MetaError> + Send + Sync>,
+    err_code_fn: ErrFn,
+    context_fn: CtxFn,
+) -> std::result::Result<SeqV<T>, ErrorCode>
+where
+    T: FromToProto + serde::de::DeserializeOwned + 'a + 'static,
+    T::PB: databend_common_protos::prost::Message + Default,
+    ErrFn: FnOnce(String) -> ErrorCode + std::marker::Copy,
+    D: Display,
+    CtxFn: FnOnce() -> D + std::marker::Copy,
+{
+    let deserialize_result =
+        deserialize_struct(&seq_value.data, ErrorCode::IllegalUserInfoFormat, || "");
+
+    let err = match deserialize_result {
+        Ok(data) => return Ok(SeqV::new(seq_value.seq, data)),
+        Err(err) => err,
+    };
+
+    log::debug!("deserialize as pb err: {}, rollback to use serde json", err);
```

Comment:
`log::info!`?

Reply:
I think 'Info' level might be too frequent. I prefer we only see information when we want to, so I chose 'Debug'.